Mage/src/main/java/mage/players/PlayerImpl.java

```diff
@@ -5140,59 +5135,85 @@ public FilterMana getPhyrexianColors() {
+            game.informPlayers(getLogName() + " has no creature to be Ring-bearer.");
             return;
         }
-        TargetPermanent target = new TargetControlledCreaturePermanent();
-        target.setNotTarget(true);
-        target.withChooseHint("to be your Ring-bearer");
-        choose(Outcome.Neutral, target, null, game);
-        UUID newBearerId = target.getFirstTarget();
-        if (game.getPermanent(newBearerId) != null) {
+
+        // There should always be a creature at the end.
+        UUID newBearerId;
+        if(ids.size() == 1){
+            // Only one creature, it will be the Ring-bearer.
+            // The player does not have to make any choice.
+            newBearerId = ids.get(0);
+        } else {
+            // Multiple possible Ring-bearer.
+            // Asking first if the player wants to change Ring-bearer.
+            boolean mustChoose = currentBearer == null || !(currentBearer.isCreature(game));
+            boolean choosing = mustChoose;
+            if (!mustChoose) {
+                choosing = chooseUse(Outcome.Neutral, "Choose a new Ring-bearer?", null, game);
+            }
+
+            if (choosing) {
+                TargetPermanent target = new TargetControlledCreaturePermanent();
+                target.setNotTarget(true);
+                target.withChooseHint("to be your Ring-bearer");
+                choose(Outcome.Neutral, target, null, game);
+
+                newBearerId = target.getFirstTarget();
+            }
+            else {
+                newBearerId = currentBearerId;
+            }
+        }
+
+        if(currentBearerId != null && currentBearerId == newBearerId) {
+            // Oracle Ruling for Call of the Ring
+            //
+            // If the creature you choose as your Ring-bearer was already your Ring-bearer,
+            // that still counts as choosing that creature as your Ring-bearer for the purpose
+            // f abilities that trigger "whenever you choose a creature as your Ring-bearer"
+            // or abilities that care about which creature was chosen as your Ring-bearer.
+            // (2023-06-16)
+            game.fireEvent(GameEvent.getEvent(GameEvent.EventType.RING_BEARER_CHOSEN, currentBearer.getId(), null, getId()));
+            game.informPlayers(getLogName() + " did not choose a new Ring-bearer. " +
+                "It is still " + currentBearer.getLogName() + ".");
+        } else {
+            Permanent ringBearer = game.getPermanent(newBearerId);
             game.fireEvent(GameEvent.getEvent(GameEvent.EventType.RING_BEARER_CHOSEN, newBearerId, null, getId()));
-            this.ringBearerId = newBearerId;
+            game.informPlayers(getLogName() + " has chosen " + ringBearer.getLogName() + " as Ring-bearer.");
+
+            // The setRingBearer method is taking care of removing
+            // the status from the current ring bearer, if existing.
+            ringBearer.setRingBearer(game, true);
         }
```

Comment:
`RING_BEARER_CHOSEN` event must be called after `setRingBearer`, so triggers will see actual state of the battlefield and ring bearers.